From ff0218a00172c4db02cac1ceb66b30b6b129dc6c Mon Sep 17 00:00:00 2001
From: Narendhar Reddy <quic_nare@quicinc.com>
Date: Thu, 13 Jul 2023 20:14:23 +0530
Subject: [PATCH] ath12k : Add crash dump and hybrid bus support for qcn6432.

Enabled crash dump support for qcn6432. Added hybrid bus support for
qcn6432 in the core dump collection mechanism.

Signed-off-by: Narendhar Reddy <quic_nare@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/ahb.c  | 2 ++
 drivers/net/wireless/ath/ath12k/core.c | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/ahb.c b/drivers/net/wireless/ath/ath12k/ahb.c
index 2228ddc..665f8a6 100644
--- a/drivers/net/wireless/ath/ath12k/ahb.c
+++ b/drivers/net/wireless/ath/ath12k/ahb.c
@@ -691,6 +691,8 @@ static const struct ath12k_hif_ops ath12k_ahb_hif_ops_qcn6432 = {
 	.config_static_window = ath12k_pcic_config_static_window,
 	.get_msi_irq = ath12k_pcic_get_msi_irq,
 	.map_service_to_pipe = ath12k_pcic_map_service_to_pipe,
+	.ssr_notifier_reg = ath12k_ahb_ssr_notifier_reg,
+	.ssr_notifier_unreg = ath12k_ahb_ssr_notifier_unreg,
 };
 
 static int ath12k_core_get_rproc(struct ath12k_base *ab)
diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 8bb0c54..547a3f3 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -3146,7 +3146,7 @@ static void ath12k_core_reset(struct work_struct *work)
 	 */
 	if (ab->hif.bus == ATH12K_BUS_PCI) {
 		ath12k_coredump_download_rddm(ab);
-	} else if (ab->hif.bus == ATH12K_BUS_AHB) {
+	} else if (ab->hif.bus == ATH12K_BUS_AHB || ab->hif.bus == ATH12K_BUS_HYBRID) {
 		ath12k_coredump_download_ahb(ab);
 	}
 	if (ab->is_qdss_tracing)
-- 
2.17.1
